lib/ReactViews/DragDropFile.tsx

```diff
@@ -50,19 +57,26 @@ class DragDropFile extends React.Component<PropsType> {
           MappableMixin.isMixedInto
         );
 
-        yield Promise.all(mappableItems.map(f => f.loadMapItems()));
+        Result.combine(
+          await Promise.all(mappableItems.map(f => f.loadMapItems())),
+          "Failed to load uploaded files"
+        ).raiseError(props.terria);
 
         // Zoom to first item
         const firstZoomableItem = mappableItems.find(i =>
           isDefined(i.rectangle)
         );
 
         isDefined(firstZoomableItem) &&
-          props.terria.currentViewer.zoomTo(firstZoomableItem, 1);
+          runInAction(() =>
+            props.terria.currentViewer.zoomTo(firstZoomableItem, 1)
+          );
       }
 
-      props.viewState.isDraggingDroppingFile = false;
-    })();
+      runInAction(() => (props.viewState.isDraggingDroppingFile = false));
+    } catch (e) {
+      props.terria.raiseErrorToUser(e, "Failed to upload files");
+    }
   }
```

Comment:
No need for `flow` here - its easier to just deal with an `async function` and use `runInAction` when needed.

Also Resultify